src/test.js

```diff
@@ -485,15 +485,14 @@ function synchronize( callback, priority ) {
 
 // Place previously failed tests on a queue priority line, respecting the order they get assigned.
 function priorityFill( callback ) {
-	var queue, prioritizedQueue;
+	var highPrioritized, lowPrioritized;
 
-	queue = config.queue.slice( priorityFill.pos );
-	prioritizedQueue = config.queue.slice( 0, -config.queue.length + priorityFill.pos );
+	highPrioritized = config.queue.splice( 0, priorityFill.pos );
+	lowPrioritized = config.queue;
+	highPrioritized.push( callback );
+	lowPrioritized.unshift.apply( lowPrioritized, highPrioritized );
 
-	queue.unshift( callback );
-	queue.unshift.apply( queue, prioritizedQueue );
-
-	config.queue = queue;
+	config.queue = lowPrioritized;
```

Comment:
This whole block looks like it could be simplified down to `config.queue.splice( priorityFill.pos, 0, callback )` (or even `config.queue.splice( priorityFill.pos++, 0, callback )` for a one-line function :wink:).